wordpress-mobile/WordPress-Android
# [Dependency Updates] Update `firebaseBomVersion` to 32.1.0

Parent #17567

This PR migrates to [BOM](https://firebase.google.com/support/release-notes/android#compare-bom-versions) and uses the `firebaseBomVersion` update as of [32.1.0](https://firebase.google.com/support/release-notes/android#bom_v32-1-0). To be more specific, this PR actually updates `firebaseMessagingVersion` to [23.1.2](https://firebase.google.com/support/release-notes/android#messaging_v23-1-2).

Also, as part of this update the below transitive dependencies were added:

- On the `WordPress` module (5c1eebab5b2ee4f09576f05a84cbc5559a2f63ed):
    - `com.google.firebase:firebase-iid` (`DEPRECATED`) ⚠️ (refer to [this](https://github.com/wordpress-mobile/WordPress-Android/issues/17229) and [this](https://github.com/wordpress-mobile/WordPress-Android/issues/18185))

-----

PS: @ravishanker  I added you as the main reviewer, randomly so, since I just wanted someone from the WordPress team to be aware of and sign-off on that change for WPAndroid. I also added the @wordpress-mobile/apps-infrastructure team, but this in done only for monitoring purposes, as such, I am not expecting any active review from that team. Thus, feel free to merge this PR if you deem so.

-----

## To test:

1. See the dependency tree diff result and verify correctness.
2. Thoroughly smoke test any Firebase messaging functionality (aka push notifications) on both, the WordPress and Jetpack apps, and see if everything is working as expected.
3. In addition to the above smoke test, you can expand the below and follow the inner and more explicitly test steps within:

<details>
    <summary>Push Notification on Post Being Published [GCMMessageService.java]</summary>

ℹ️ This test applies to both, the `WordPress` and `Jetpack` apps.

- Add a new `blog` post.
- Add any number of blocks.
- Publish the post.
- Verify that, on post being published, you get a push notification with the post's title, which, when you click on it, navigates you to the 'Notifications Detail' screen for this post.

</details>

FYI: You could also use this [Push Notifications Debug Console](https://mc.a8c.com/mobile/push-notifications/debug/) tool to verify that push notifications work as expected.

-----

## Regression Notes

1. Potential unintended areas of impact

    - Potential breakage or misbehaviour on any or all Firebase related messaging functionalities (aka push notifications).
    - Potential breakage of misbehaviour of how Firebase tokens are being renewed (refer to [this](https://github.com/wordpress-mobile/WordPress-Android/blob/f5b756308b107d59f5ea0c4f7bae70a8c6fedd97/WordPress/src/main/java/org/wordpress/android/push/GCMMessageService.java#L99-L103)) and how the Firebase instance ID is getting deleted (refer to [this](https://github.com/wordpress-mobile/WordPress-Android/blob/f5b756308b107d59f5ea0c4f7bae70a8c6fedd97/WordPress/src/main/java/org/wordpress/android/AppInitializer.kt#L633-L639)).

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:


-----

UI Changes testing checklist:
